eventio/event_io_file.py

```diff
@@ -0,0 +1,65 @@
+import struct
+from .object_header import make_ObjectHeader
+from io import BytesIO
+
+class EventIOObject:
+    ''' A generic EventIOObject
+
+    It has a list of `headers` and a binary string `payload`.
+    The payload might be loaded lazily on first access or already in memory.
+    This is decided on construction time.
+    '''
+    def __init__(self, headers, payload=None, file=None):
+        if file is not None:
+            self._file = file
+        elif payload is not None:
+            self._file = None
+            self.payload = payload
+        else:
+            raise Exception('EventIOObject must have either file or payload')
+        
+        self.headers = headers
+
+    def __getattr__(self, attr):
+        if attr == "payload":
+            if self._file.closed:
+                self._file = open(self._file.name, 'rb')
```

Comment:
Yup. If every object keeps a reference to the file, the file will only be closed if the user explicitly calls `file.close()`.